uncertainty_engine/api_providers/resource_provider.py

```diff
@@ -365,25 +363,27 @@ def list_resources(
             A list of all matching resources with their details
 
         Example:
-            >>> models = client.list(
+            >>> models = client.resources.list_resources(
             ...     project_id="your-project-123",
             ...     resource_type="model"
             ... )
             >>> print(f"Found {len(models)} models")
             >>> print(models)
         """
-        resource_records = self.resources_client.get_project_resource_records(
-            project_id, resource_type
-        ).resource_records
-
-        return [
-            {
-                "id": record.id,
-                "name": record.name,
-                "created_at": record.created_at.strftime(DATETIME_STRING_FORMAT),
-            }
-            for record in resource_records
-        ]
+
+        try:
+
+            resource_records = self.resources_client.get_project_resource_records(
+                project_id, resource_type
+            ).resource_records
+            return [
+                ResourceRecordOutput.model_validate(record)
+                for record in resource_records
+            ]
+        except ApiException as e:
+            raise Exception(f"Error listing project resources: {format_api_error(e)}")
+        except Exception as e:
+            raise Exception(f"Error listing project resources: {str(e)}")
```

Comment:
Feel like we can remove `project` in the error message here (unless you disagree) will mean you have to update corresponding test regex as well.
```suggestion
        except ApiException as e:
            raise Exception(f"Error listing resource records: {format_api_error(e)}")
        except Exception as e:
            raise Exception(f"Error listing resource records: {str(e)}")
```